packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.ts

```diff
@@ -432,13 +448,14 @@ export function useShopperBasketsMutation<Action extends ShopperBasketMutationTy
     const customerId = useCustomerId()
     const cacheUpdateMatrix = getCacheUpdateMatrix(customerId)
 
-    type Params = Argument<Client[Action]>
-    type Data = DataType<Client[Action]>
-
     return useMutation<Data, Error, Params>(
         (params, apiClients) => {
             const method = apiClients['shopperBaskets'][action] as MutationFunction<Data, Params>
-            return method.call(apiClients['shopperBaskets'], params)
+            //Override param headers to user-defined header values
+            if (params) {
+                params.headers = headers
+            }
+            return (method.call as any)(apiClients['shopperBaskets'], params, rawResponse)
```

Comment:
Can you create a type (or use an existing one) for this with the expected signature `apiClients['shopperBaskets'], params, rawResponse` something like

```
type ShopperBasketsMutationParams = {
  param1: (whatever type is for apiClients['shopperBaskets']);
  param2:(whatever type is for `params`); 
  param3: (whatever type is for `rawResponse`);
}
```

and then

```
return (method.call as ShopperBasketsMutationParams)(apiClients['shopperBaskets'], params, rawResponse)
```

This should be **simple and non-dynamic typing**